src/gostudy.py

```diff
@@ -1,97 +1,74 @@
-                 db = client.IGCSEBot
-                 mute = db["mute"]
-                 mute.insert_one({"_id": str(timern), "user_id": str(user_id),
-                                  "unmute_time": str(unmute_time), "muted": True})
-            proceedBTN.callback = proceedCallBack
-            async def cancelCallBack(interaction):
-                 await message.delete()
-            cancelBTN.callback = cancelCallBack
-            view.add_item(proceedBTN)
-            view.add_item(cancelBTN)
-            message = await interaction.send("Are we ready to move forward?", view=view, ephemeral=True)
-
-@bot.slash_command(name="remove_gostudy", description="remove the Forced Mute role.")
-async def remove_gostudy(interaction: discord.Interaction,
-                  user: discord.User = discord.SlashOption(name="name", description="who do you want to use this command on?", required=False)):
-        await interaction.response.defer(ephemeral = True)
-        if not await is_moderator(interaction.user) and not await has_role(interaction.user, "Bot Developer"):
-                  await interaction.send("You do not have the necessary permissions to perform this action", ephemeral = True)
-                  return
-        
-        forced_mute_role = bot.get_guild(GUILD_ID).get_role(FMROLE)
-        if user == None:
-            user_id = interaction.user.id
-            guild = bot.get_guild(GUILD_ID)
-            member = guild.get_member(user_id)
-            forced_mute_role = bot.get_guild(GUILD_ID).get_role(FMROLE)
-            await member.remove_roles(forced_mute_role)
-            client = pymongo.MongoClient(LINK)
-            db = client.IGCSEBot
-            mute = db["mute"]
-            mute.delete_one({"user_id": str(user_id)})
-            await interaction.send(f"the Forced mute role for <@{user_id}> has been removed", ephemeral=True)
-
-        else:
-            user_id = user.id
-            user = bot.get_guild(GUILD_ID).get_member(user_id)
-            guild = bot.get_guild(GUILD_ID)
-            member = guild.get_member(user_id)
-            forced_mute_role = bot.get_guild(GUILD_ID).get_role(FMROLE)
-            await member.remove_roles(forced_mute_role)
-            client = pymongo.MongoClient(LINK)
-            db = client.IGCSEBot
-            mute = db["mute"]
-            mute.delete_one({"user_id": str(user_id)})
-            await interaction.send(f"the Forced mute role for <@{user_id}> has been removed", ephemeral=True)
+
+
+@bot.slash_command(name="gostudy", description="Disables access to the offtopic channels for the next 1 hour.")
+async def gostudy(
+    interaction: discord.Interaction,
+    user: discord.User = discord.SlashOption(
+        name="name", description="Who do you want to use this command on? (only for Mods)", required=False
+    ),
+):
+    guild = bot.get_guild(GUILD_ID)
+    forced_mute_role = guild.get_role(FMROLE)
+
+    user_id = interaction.user.id if user is None else user.id
```

Comment:
In this updated `/gostudy` command code there's no isModerator check when choosing a particular user so anyone would be able to Study Mute another member.